### PR TITLE
Checking if a file is a res file

### PR DESCRIPTION
**Pull request by zth**
This is a continuation of https://github.com/rescript-lang/rescript-vscode/pull/539

### CODE REVIEW

File: analysis/src/Hint.ml [hunk 1]
```diff
@@ -82,7 +82,7 @@ let inlay ~path ~pos ~maxLength ~debug =
     Ast_iterator.default_iterator.value_binding iterator vb
   in
   let iterator = {Ast_iterator.default_iterator with value_binding} in
-  (if Files.exists path && Filename.check_suffix path ".res" then
+  (if Files.isResFile path then
```
Comment (cristianoc) [inline analysis/src/Hint.ml:85]: I'm not sure that this does it.
The temptation could be `else ...` for `.resi` files which would leak to a crash.
Any other ideas?

Comment (zth) [reply analysis/src/Hint.ml]: Check the latest commit, what do you think about that? Slightly harder at to accidentally do the wrong thing at least I think, but still not fool proof.

**Review by cristianoc — APPROVED** [review]
Great?
"classifySourceFile"

**Comment by zth** [comment]
Fixed the name! Merging now.

**Pull request merged**